### PR TITLE
Refactor and polish iterations 0 through 2

### PR DESCRIPTION
**Pull request by knees4bees**
## **_PR TEMPLATE_**
1. Is this a fix or a feature?
* Fix: mostly refactoring and cleaning up but also reinstated a few details we lost in a confusing merge situation on Thursday evening.
* 
2. What is the change?
* Some new html to reinstate alt and placeholder text lost in the merge; new show and hide functions in js; general comment cleanup.
* 
3. What does it fix?
* Reinstates code lost in the confusing merge.
* 
4. Where should the reviewer start?
* Clone down the repo and open `index.html` in your browser.

5. How should this be tested?
* Confirm that the app allows you to select one (and only one) activity category, requires you to fill out all other form fields appropriately, and then upon click of the "Start Activity" button, a circle(ish) with a countdown timer appears.
* For reference, project spec with comp screenshots can be found at https://frontend.turing.io/projects/module-1/intention-timer-group.html (we're currently working on iteration 2, which is almost complete except for displaying the user-submitted time on the countdown clock).
* We welcome comments on all aspects of our code so far, and thank you for reviewing it. :)

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -26,136 +26,99 @@ categoryBoxWrapper.addEventListener('click', activateCategory);
...
 //Add functions below
+function show(element) {
+  element.classList.remove('hidden');
+}
+
+function hide(element) {
+  element.classList.add('hidden');
+}
+
 function submitForm() {
```
Comment (stevecalla) [inline main.js]: Need to pass in "event" as a parameter when using event.preventDefault.

File: main.js [hunk 2]
```diff
@@ -26,136 +26,99 @@ categoryBoxWrapper.addEventListener('click', activateCategory);
 startActivityButton.addEventListener('click', submitForm);
 
 //Add functions below
+function show(element) {
```
Comment (stevecalla) [inline main.js:29]: Let's establish an order (if we haven't already) to the function list. My recommendation. (a) order based on event listener so all categoryBoxWrapper functions are first, (b) then order based on the sequence in the function called in the event listener... so for startActivityButton... 1) subitForm, 2) createInputValidity, 3) createNewActivity, 4) and so on following the functions as they are used...

File: index.html [hunk 3]
```diff
@@ -17,47 +17,45 @@ <h1 class="title">INTENTION TIMER</h1>
         <h2 class="subhead">New Activity</h2>
           <section class="category-timer-wrapper">
             <article class="timer-box-wrapper hidden">
-            <!-- <article class="timer-box-wrapper"> -->
               <h3 class="timer-box-header">Deep Breathing</h3>
-              <h1 class="countdown-clock">05:00</h2>
+              <h1 class="countdown-clock">05:00</h1>
```
Comment (stevecalla) [inline index.html:21]: thumbs up!

File: index.html [hunk 4]
```diff
@@ -17,47 +17,45 @@ <h1 class="title">INTENTION TIMER</h1>
...
                 <p class="start-circle-text">START</p>
               </div>
             </article>
-            <!-- <form class="form hidden"> -->
             <form class="form">
               <article class="categories">
                 <h3 class="form-label">Select a category:</h3>
                 <section class="category-box-wrapper">
                   <div class="category-boxes study-box">
                     <img class="category-image study-image" src="./assets/study.svg" alt="light bulb above an open book">
```
Comment (stevecalla) [inline index.html]: Might consider changing consistent with "Option 1 - select study activity", "Option 2 - select meditate activity", et al.

File: index.html [hunk 5]
```diff
@@ -17,47 +17,45 @@ <h1 class="title">INTENTION TIMER</h1>
...
-                    <img class="category-image study-image-active hidden" src="./assets/study-active.svg" alt="green light bulb above an open book">
+                    <img class="category-image study-image-active hidden" src="./assets/study-active.svg" alt="study activity selected">
                     <p class="category-text">Study</p>
```
Comment (stevecalla) [inline index.html:33]: Might consider changing consistent with "Option 1 - select study activity", "Option 2 - select meditate activity", et al.

File: index.html [hunk 6]
```diff
@@ -17,47 +17,45 @@ <h1 class="title">INTENTION TIMER</h1>
...
                   </div>
                   <div class="category-boxes meditate-box">
                     <img class="category-image meditate-image" src="./assets/meditate.svg" alt="lotus flower">
```
Comment (stevecalla) [inline index.html]: Might consider changing consistent with "Option 1 - select study activity", "Option 2 - select meditate activity", et al.

File: main.js [hunk 7]
```diff
@@ -26,136 +26,99 @@ categoryBoxWrapper.addEventListener('click', activateCategory);
...
+  element.classList.add('hidden');
+}
+
 function submitForm() {
   event.preventDefault();
   if (checkInputValidity()) {
     createNewActivity();
-  };
-};
+    hide(activityInputForm);
```
Comment (stevecalla) [inline main.js:41]: Nice!! We did lose this during that merge b/c we replaced it with refactored code.

File: main.js [hunk 8]
```diff
@@ -26,136 +26,99 @@ categoryBoxWrapper.addEventListener('click', activateCategory);
...
+    show(timerBoxWrapper);
+  }
+}
 
 function checkInputValidity() {
   //Add validation conditional here for activity boxes to make sure one is clicked
-  if (!accomplishInput.value === true) {
+  if (!accomplishInput.value) {
```
Comment (stevecalla) [inline main.js:48]: Nice. Jessica and I discussed and she tried but couldn't get it to work without strictly equals. Good deal!

File: main.js [hunk 9]
```diff
@@ -26,136 +26,99 @@ categoryBoxWrapper.addEventListener('click', activateCategory);
...
+  }
+}
 
 function createNewActivity() {
   //"Exercise" is a placeholder for category box input
   currentActivity = new Activity("Exercise", accomplishInput.value, minutesInput.value, secondsInput.value);
-  console.log(currentActivity);
   //save activities in localStorage or array??
-};
+}
```
Comment (stevecalla) [inline main.js:66]: Really, Really!!

File: main.js [hunk 10]
```diff
@@ -26,136 +26,99 @@ categoryBoxWrapper.addEventListener('click', activateCategory);
...
   studyImageActive.classList.toggle('hidden');
-  studyBox.classList.toggle('study-border');
-  meditateImage.classList.remove('hidden');
-  meditateImageActive.classList.add('hidden');
-  meditateBox.classList.remove('meditate-border');
-  exerciseImage.classList.remove('hidden');
-  exerciseImageActive.classList.add('hidden');
-  exerciseBox.classList.remove('exercise-border');
+  studyBox.classList.toggle('study-active');
+  show(meditateImage);
```
Comment (stevecalla) [inline main.js:88]: Nice reduction!

File: styles.css [hunk 11]
```diff
@@ -124,6 +124,7 @@ form {
 .category-image {
   width: 1.5em;
   margin-top: 10px;
+  pointer-events: none;
```
Comment (stevecalla) [inline styles.css:127]: Wowza!

File: styles.css [hunk 12]
```diff
@@ -132,17 +133,17 @@ form {
   pointer-events: none;
 }
 
-.study-border {
+.study-active {
```
Comment (stevecalla) [inline styles.css:136]: Yes. Name represents multiple changes happening for color on the text and border (not only border).

File: main.js [hunk 13]
```diff
@@ -26,136 +26,99 @@ categoryBoxWrapper.addEventListener('click', activateCategory);
...
+  if (!accomplishInput.value) {
      alert('Please enter description');
      return false;
-  } else if (!minutesInput.value === true || isNaN(minutesInput.value)) {
+  } else if (!minutesInput.value || isNaN(minutesInput.value)) {
      alert('Please enter valid minutes input');
      return false;
-  } else if (!secondsInput.value === true || isNaN(secondsInput.value) || secondsInput.value > 59) {
+  } else if (!secondsInput.value || isNaN(secondsInput.value) || secondsInput.value > 59) {
      alert(`Please enter valid seconds input`);
```
Comment (stevecalla) [inline main.js:55]: @m1073496 I believe we lost some error message coding here or in the HTML/CSS to show an error message below the description, minutes and seconds? Can we redo?

File: main.js (hunk 1 above)
Comment (knees4bees) [reply main.js]: I think I see what you're getting at but I don't think it applies in this case. Here `element` is not an event; it's a placeholder parameter that just means "html element".

File: main.js (hunk 2 above)
Comment (knees4bees) [reply main.js:29]: Sounds good to me!

Comment (knees4bees) [reply main.js:29]: I added a card for this on the project board

File: main.js (hunk 1 above)
Comment (knees4bees) [reply main.js]: Ohhhhh nevermind, this GH interface confused me and I thought you were talking about line 33, not line 37. Got it, and agreed. I'll change and push now/soon.

File: styles.css (hunk 11 above)
Comment (m1073496) [reply styles.css:127]: 👌🏻

File: main.js (hunk 13 above)
Comment (m1073496) [reply main.js:55]: @stevecalla I didn't get a chance to reroute the alert messages to the spots below the description. I think we should fix up the CSS on those alert icons/messages first though.

File: main.js (hunk 8 above)
Comment (m1073496) [reply main.js:48]: Nice catch here @knees4bees! I think I was trying `=== false` at first, then switched to true to get the functionality I needed, but didn't realize that the `=== true` was now redundant.

File: index.html (hunk 4 above)
Comment (knees4bees) [reply index.html]: Right on. Done and pushed.

File: main.js (hunk 13 above)
Comment (knees4bees) [reply main.js:55]: on project board

**Comment by m1073496** [comment]
Looks great!

**Pull request merged**